ssl/s3_lib.c

```diff
@@ -4986,3 +4971,12 @@ int ssl_encapsulate(SSL *s, EVP_PKEY *pubkey,
     EVP_PKEY_CTX_free(pctx);
     return rv;
 }
+
+const char* SSL_groupid_to_name(SSL *s, int group_id) {
+    const TLS_GROUP_INFO *cinf = tls1_group_id_lookup(s->ctx, group_id);
```

Comment:
This works by accident. `tls1_group_id_lookup` takes a `uint16_t` as its second parameter. Here you are doing an implicit cast from `int` to `uint16_t`. That means if the input value happens to have the `TLSEXT_nid_unknown` bit set then that bit gets dropped and it works. This is why your test works. But it won't work for real nids - which it should do.

I think we should introduce a new internal function `tls1_nid2group` which works a bit like `tls1_nid2group_id` but returns a `TLS_GROUP_INFO *` and is able to handle the `TLSEXT_nid_unknown` semantics.

Reply:
Agreed. Please check new code as to whether it correctly captures the above. I refrained from adding another function as `tls1_nid2group_id` could be utilized suitably.

Reply:
Addressed your correct comments with the update - in a slightly different way (not adding a new function, but using an existing one). 

A final "etiquette" question: Please let me know whether I should click "Resolve conversation" if I think I followed through or whether the person raising the objection will do that after reviewing my comments/code changes.

Reply:
Feel free to click "resolve" if you believe that you've fully fixed an issue. Rest assured that a reviewer will "unresolve" if they disagree :-)

Reply:
> A final "etiquette" question: Please let me know whether I should click "Resolve conversation" if I think I followed through or whether the person raising the objection will do that after reviewing my comments/code changes.

I don't think we have a codified practice on this, my personal rule of thumb is that, as the PR author, I tend to click "Resolve conversation" myself only in the trivial ones (e.g. "there is a missing space here", "typo there") after writing a "Done/Fixed/Addressed" comment.
Instead, I leave it to whomever started the discussion for the other cases, especially if I'd like to have some explicit confirmation that my fix fully addressed the issue as intended by the reviewer.

But again, this is just my rule of thumb.